src/MongoDB.Driver.Core/Core/Servers/ServerDescription.cs

```diff
@@ -109,13 +115,15 @@ public sealed class ServerDescription : IEquatable<ServerDescription>
             _endPoint = endPoint;
             _heartbeatException = heartbeatException.WithDefault(null);
             _heartbeatInterval = heartbeatInterval.WithDefault(TimeSpan.Zero);
+            _lastHeartbeatTimestamp = lastHeartbeatTimestamp.WithDefault(null);
             _lastUpdateTimestamp = lastUpdateTimestamp.WithDefault(DateTime.UtcNow);
             _lastWriteTimestamp = lastWriteTimestamp.WithDefault(null);
             _logicalSessionTimeout = logicalSessionTimeout.WithDefault(null);
             _maxBatchCount = maxBatchCount.WithDefault(1000);
             _maxDocumentSize = maxDocumentSize.WithDefault(4 * 1024 * 1024);
             _maxMessageSize = maxMessageSize.WithDefault(Math.Max(_maxDocumentSize + 1024, 16000000));
             _maxWireDocumentSize = maxWireDocumentSize.WithDefault(_maxDocumentSize + 16 * 1024);
+            _reasonChanged = reasonChanged.WithDefault("NotSpecified");
```

Comment:
Can it be just null? When we just create a ServerDescription first time, the reasonChanged field makes no sense to me

Reply:
Also, not sure about naming, when we create it the first time, there is nothing to change. Maybe the name like: `triggerReason` will be more accurate? But I don't insist. I will be ok with the current name too

Reply:
I wouldn't worry too much about the first reasonChanged. Let's focus on making the thousands of subsequent changes readable.

Reply:
ok